### ISSUE TITLE
Add builder for LSP code actions

### ISSUE THREAD

**Issue by kpodsiad** [scalameta/metals#4154]
Currently, building LSP code actions is quite verbose and repetitive, e.g. there are a lot of `asJava` calls needed.

```scala
      val codeAction = new l.CodeAction()
      codeAction.setTitle(CreateNewSymbol.title(name))
      codeAction.setKind(l.CodeActionKind.QuickFix)
      codeAction.setDiagnostics(List(diagnostic).asJava)
      ...
```

We can consider refactoring some builder/helper to reduce amount of boilerplate. Something along those lines should do the work.

```scala
object CodeActionBuilder {
  def build(
      title: String,
      kind: String,
      workspaceEdits: Seq[(AbsolutePath, Seq[l.TextEdit])] = Nil,
      // ... other parameters with empty defaults
  ): l.CodeAction = {
    val codeAction = new l.CodeAction()
    codeAction.setTitle(title)
    codeAction.setKind(kind)
    codeAction.setEdit(
      new l.WorkspaceEdit(
        workspaceEdits
          .map { case (path, edits) =>
            path.toURI.toString -> edits.asJava
          }
          .toMap
          .asJava
      )
    )
    codeAction
  }
}
```

**Comment by PeuTit** [comment]
Hey! I'll take this issue.

**Comment by robmwalsh** [comment]
Should this be closed now?

**Comment by ckipp01** [comment]
> Should this be closed now?

Yes, thanks for catching this @robmwalsh.

Closed via #4302
